lib/pixel-mapper.cc

```diff
@@ -217,6 +217,50 @@ class UArrangementMapper : public PixelMapper {
   int parallel_;
 };
 
+
+
+class VerticalMapper : public PixelMapper {
+public:
+  VerticalMapper() {}
+
+  virtual const char *GetName() const { return "V-mapper"; }
+
+  virtual bool SetParameters(int chain, int parallel, const char *param) {
+    chain_ = chain;
+    parallel_ = parallel;
+    return true;
+  }
+
+  virtual bool GetSizeMapping(int matrix_width, int matrix_height,
+                              int *visible_width, int *visible_height)
+    const {
+    *visible_width = matrix_width * parallel_ / chain_;
+    *visible_height = matrix_height * chain_ / parallel_;
+      fprintf(stderr, "%s: C:%d P:%d. Turning W:%d H:%d Physical "
+	      "into W:%d H:%d Virtual\n",
+              GetName(), chain_, parallel_,
+	      *visible_width, *visible_height, matrix_width, matrix_height);
+    return true;
+  }
+
+  virtual void MapVisibleToMatrix(int matrix_width, int matrix_height,
+                                  int x, int y,
+                                  int *matrix_x, int *matrix_y) const {
+    int panel_width  = matrix_width  / chain_;
+    int panel_height = matrix_height / parallel_;
+    *matrix_x = (x % panel_width) +  int(y/panel_height)* panel_width;
+    *matrix_y = (y % panel_height) + int(x/panel_width) * panel_height;
+
+    //fprintf(stderr, "%s: Panel-W:%d Panel-H:%d. X: %3d -> %3d, Y: %3d -> %3d\n",
```

Comment:
Don't leave commented-out code in the final version.

Reply:
So, I left it there on purpose it case it helps someone (possibly me) writing the next mapper. Understanding how things worked without it would have been hard (it was non obvious to see how things were mapped)
If you'd prefer, I can put it under an #ifdef 0  or something else, though, but if you really don't want it, I can remove it and apply it again in my tree as I will definitely want it again in the future if I write another mapper.
What's your call?

Reply:
Removing commented-out here. Someone who will want to debug this will add something very similar in the working copy, but here, we should leave out the clutter.

Also above in the GetSizeMapping(), the output should not be there, but you can `#if 0` that out if you want.

Reply:
Ok, PTAL